src/irmin-pack/conf.mli

```diff
@@ -27,7 +27,11 @@ module type S = sig
       - [Some `Varint]: the length header is a LEB128-encoded integer at the
         very beginning of the encoded value.
 
-      - [None]: there is no length header, and values have unknown size. *)
+      - [None]: there is no length header, and values have unknown size. NOTE:
```

Comment:
I'm assuming we expect all future objects in the store to use Varint length encoding? Or some other way to indicate length (a tag indicating the type of the encoded object, which is then of fixed length)? If every object will use Varint in the future, perhaps we should add a line to say "Use of None is deprecated and is supported only for historical usecases" or something similar, just to emphasize that we are moving away from this.

Reply:
I think for now at least it's reasonable to continue supporting contents values with opaque length information (e.g. fixed size, stored externally, stored in some format unknown to `irmin-pack`), since as you say we already support this behaviour for older internal objects. I'd probably wait until we have a motive before deprecating that, at any rate.